relay-server/src/services/processor.rs

```diff
@@ -1616,38 +1638,35 @@ impl EnvelopeProcessorService {
         let clock_drift_processor =
             ClockDriftProcessor::new(sent_at, received).at_least(MINIMUM_CLOCK_DRIFT);
 
+        let mut buckets: Option<Vec<Bucket>> = None;
```

Comment:
Would this work without the option? An empty vector does not allocate.

```suggestion
        let mut buckets = vec![];
```